### ISSUE TITLE
User statistics or survey

### ISSUE THREAD

**Issue by apjanke** [octave-app/octave-app#209]
It would be really helpful to have some info on our users, what systems they're running, and what they use Octave.app for and with, to guide our design choices and development priorities. I'd really like to have something like breakdowns of:

* What macOS version are they using?
* What Octave major version do they want to be running?
* Are they installing Octave Forge packages, and if so, which ones?
* Are they running in to errors?
* How many users do we have, anyway?

The macOS version is the most important one; that will guide us in choosing when to drop support for older macOS versions, which is important for us logistically.

I'm not willing to add any user tracking or telemetry to Octave.app or do download analytics (and I think the GNU Octave folks would be pissed at me if I did). So this has to come from community outreach.

How can we figure out who our users are, what they're doing, and what they want?

I posted https://github.com/octave-app/octave-app/issues/25 a while back, and that got all of four replies. So, people aren't watching our GitHub Issues actively.

Could we:

* Create a user mailing list?
  * Or maybe a forum somewhere?
    * Or maybe a subtopic in the main Octave Discord forum?
  * And advertise that mailing list or forum in the first-run splash page for Octave.app?
* Add a section to the [Octave.app web site](https://octave-app.org) begging users to contact us and tell us who they are and what they're running on?
* Use our sad, zero-follower [OctaveDotApp Twitter account](https://twitter.com/octavedotapp) in some way?
* Have a first-run splash page that invites users to just email us and tell us who they are and what they're running?

**Comment by jrminter** [comment]
I like the github idea. The other thing you can do is take a look at the Fiji/imageJ forum  here: https://forum.image.sc/  for ideas... Thanks again for your work on this. I would guess that if people were having significant problems, you would have heard about it...

best regards,
John

**Comment by apjanke** [comment]
Thanks! I'll have a look through that.

Yeah, the big thing is, I largely want to know about what macOS versions our users are running, so we can decide when to discontinue support for older versions. And I'd like to do so kind of aggressively, for work and cost reasons. It's not so much about problems users are having now, it's about problems I'm going to cause for them in the future... :(

And I'd also like to know if our users are using a bunch of Octave Forge packages, and what their version/upgrade needs are, so we can decide if we want to do the (substantial) work to bundle all or some of the Forge packages inside the main Octave.app distribution. https://github.com/octave-app/octave-app/issues/109

**Comment by jrminter** [comment]
I'm running macOS 10.15.6

**Comment by rdzman** [comment]
I tend to keep up with the latest macOS versions. Any older machines that can't upgrade the OS I tend to use for testing older software versions. So for me, supporting older OS's with new builds isn't critical. It would actually be more useful to me to be able to install older Octave versions on new macOS versions.

I suggest setting up a low traffic e-mail list that is used primarily for announcing new versions, something that you can expect most actual users to want to sign up for. If you want feedback on something, then you can send out a notice with a link to a GitHub issue where people can respond. Just an idea.

**Comment by apjanke** [comment]
Closing this as hopeless, until and unless we get an actual Octave.app mailing list and/or forum up and running, which is unlikely to happen any time soon, given our small user base. And I am definitely not going to install any sort of "analytics" or user-tracking stuff in to the Octave.app builds.

Instead of a mailing list, I have set up two "announcements" tickets here on the GitHub issue tracker. I post release announcements etc. as comments on them, so users can subscribe to them, and they'll get alert emails from GitHub whenever I do so.

* https://github.com/octave-app/octave-app/issues/1
* https://github.com/octave-app/octave-app/issues/289
